components/raftstore/src/store/worker/pd.rs

```diff
@@ -525,71 +526,72 @@ where
         }
         let router = self.router.clone();
         let scheduler = self.scheduler.clone();
-        let f = self
-            .pd_client
-            .ask_batch_split(region.clone(), split_keys.len())
-            .then(move |resp| {
-                match resp {
-                    Ok(mut resp) => {
-                        info!(
-                            "try to batch split region";
-                            "region_id" => region.get_id(),
-                            "new_region_ids" => ?resp.get_ids(),
-                            "region" => ?region,
-                            "task" => task,
-                        );
+        let f = Compat::new(
```

Comment:
You can make the chain into async closure.

Reply:
For this PR, this is irrelevant, and upgrading to raftstore later will modify this
https://github.com/tikv/tikv/issues/8095